File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
  let answer = [];

  const scores = new Map();
  for (let i = 0; i < name.length; i++) {
    scores.set(name[i], yearning[i]);
  }

  photo.forEach((names) => {
    const sum = names.reduce((s, v) => {
      const score = scores.get(v) || 0;
      return s + score;
    }, 0);
    answer.push(sum);
  });

  return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    yearning_dict = {n : y for n, y in zip(name, yearning)}
    result = []
    
    for p in photo:
        score = 0
        
        for person in p:
            score += yearning_dict.get(person, 0)
            
        result.append(score)
    
    return result
